modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java

```diff
@@ -122,6 +127,32 @@ public void processOpts() {
         }
     }
 
+    @Override
+    public Map<String, Object> postProcessModels(Map<String, Object> objs) {
+        objs = super.postProcessModels(objs);
+        List<Map<String, String>> imports = (List<Map<String, String>>) objs.get("imports");
+
+        boolean addedErrorsImport = false;
+        List<Map<String, Object>> models = (List<Map<String, Object>>) objs.get("models");
+        for (Map<String, Object> m : models) {
+            Object v = m.get("model");
+            if (v instanceof CodegenModel) {
+                CodegenModel model = (CodegenModel) v;
+                if (!model.isEnum) {
+                    imports.add(createMapping("import", "encoding/json"));
```

Comment:
This and the import below have a potential to break in the case that someone invoke with `--import-mappings=import=encoding/json,import=error` (or however the mappings are done for golang). See CodegenConfigurator line 540.

Reply:
Ok, thanks, I'll take a look.

Reply:
But the same argument would apply to current `AbstractGoCodegen.postProcessModels` which adds some imports too, right? So it's a generic issue that's already there AFAICS. (And I think the best course of action would be to deduplicate the imports list after processing is done, but I think that's out of scope of this PR).